src/dbus_api/api/manager_2_0/methods.rs

```diff
@@ -100,7 +101,11 @@ pub fn destroy_pool(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
         }
     };
 
-    let msg = match dbus_context.engine.borrow_mut().destroy_pool(pool_uuid) {
+    let mut engine = dbus_context.engine.borrow_mut();
+
+    let result = pre_pool_op! {"preparing to destroy pool with uuid {}", pool_uuid; "successfully destroyed pool with uuid {}"; engine.destroy_pool(pool_uuid)};
```

Comment:
Please use the macro!() form as opposed to the macro!{} form. While these are identical from a code compilation perspective, the convention is usually to use the macro!() form for something that acts vaguely like a function. I'd argue that this acts more like a function than another construct as it simply adds some logging around the call we pass in.

Reply:
Resolved.